cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -82,30 +75,17 @@
     .registerTypeAdapter(DatasetSpecification.class, new DatasetSpecificationAdapter())
     .create();
 
-  private final DatasetTypeManager implManager;
-  private final DatasetInstanceManager instanceManager;
-  private final DatasetOpExecutor opExecutorClient;
-  private final ExploreFacade exploreFacade;
-  private final boolean allowDatasetUncheckedUpgrade;
-  private final UsageRegistry usageRegistry;
+  private final DatasetInstanceService instanceService;
 
   @Inject
-  public DatasetInstanceHandler(DatasetTypeManager implManager, DatasetInstanceManager instanceManager,
-                                DatasetOpExecutor opExecutorClient, ExploreFacade exploreFacade, CConfiguration conf,
-                                UsageRegistry usageRegistry) {
-    this.opExecutorClient = opExecutorClient;
-    this.implManager = implManager;
-    this.instanceManager = instanceManager;
-    this.exploreFacade = exploreFacade;
-    this.usageRegistry = usageRegistry;
-    this.allowDatasetUncheckedUpgrade = conf.getBoolean(Constants.Dataset.DATASET_UNCHECKED_UPGRADE);
+  public DatasetInstanceHandler(DatasetInstanceService instanceService) {
+    this.instanceService = instanceService;
   }
 
   @GET
```

Comment:
May I kindly ask that this class gets some Javadocs. I think it is important to document, for each handler method, what its path and query parameters mean.